### PR TITLE
Stricter params to prevent triggering NoMethodErrors

### PR DESCRIPTION
**Pull request by martinemde**

### CODE REVIEW

File: app/controllers/api/v1/hook_relay_controller.rb [hunk 1]
```diff
@@ -1,11 +1,10 @@
 class Api::V1::HookRelayController < Api::BaseController
-  before_action :set_hook_relay_report_params
```
Comment (martinemde) [inline app/controllers/api/v1/hook_relay_controller.rb:2]: I felt compelled to clean this controller up a bit. Sorry for the extra review overhead.

File: app/controllers/concerns/params_require.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  # params_require_hash(:user)
+  # => params.require(:user) if params[:user].respond_to?(:permit)
+  # raises ActionController::ParameterMissing if params[:user] is not a Hash
+  def params_require_hash(key, parameters = params)
+    value = parameters.require(key)
+    return value if value.respond_to?(:permit)
+    raise ActionController::ParameterMissing, key, parameters.keys
+  end
+
+  # Safely require an Arrayl parameter without crashing on bad input.
```
Comment (segiddins) [inline app/controllers/concerns/params_require.rb:17]:
```suggestion
  # Safely require an Array parameter without crashing on bad input.
```

**Review by segiddins — APPROVED** [review]

File: app/controllers/concerns/params_require.rb (hunk 2 above)
Comment (martinemde) [reply app/controllers/concerns/params_require.rb]: I ended up removing this whole module, so this is fixed.

**Comment by martinemde** [comment]
@segiddins I changed this pretty significantly. Re-requesting your review.

I threw out the module and replaced it with `params_fetch` that does everything we need and allowed me to clean up more than just the main ones.

I also made #4588, which uses params_fetch for everything. It's based on this PR. I like it, but I didn't want to block this PR on it because I think it might cross the line into non-obvious and confusing.

**Review by segiddins — APPROVED** [review]

**Pull request merged**